controllers/llamaCPP.cc

```diff
@@ -536,31 +536,40 @@ void llamaCPP::modelStatus(
 void llamaCPP::loadModel(
     const HttpRequestPtr &req,
     std::function<void(const HttpResponsePtr &)> &&callback) {
-  if (llama.model_loaded_external) {
-    LOG_INFO << "model loaded";
-    Json::Value jsonResp;
-    jsonResp["message"] = "Model already loaded";
-    auto resp = nitro_utils::nitroHttpJsonResponse(jsonResp);
-    resp->setStatusCode(drogon::k409Conflict);
-    callback(resp);
+
+  if(llama.model_loaded_external.load(std::memory_order_acquire)){
+    model_loaded_response(req, callback);
     return;
   }
+  bool modelLoadedSuccess;
+  const auto& jsonBody = req->getJsonObject();
+
+  {
+    std::scoped_lock lck{load_model_mutex};
```

Comment:
lock_guard is good enough with this single lock

Reply:
Hi,for this specific case, both work fine.
In general, if you don't have compatibility requirements, my suggestion is stick with scoped version, better api, better name and one-less-thing to remember.

Reply:
if you set for C++11, then lock_guard is only choice, totally agree.

Reply:
Thank you